src/java/org/apache/cassandra/io/sstable/format/SortedTableWriter.java

```diff
@@ -324,6 +325,19 @@ protected FileHandle openDataFile(long lengthOverride, StatsMetadata statsMetada
         return dataFile;
     }
 
+    private void guardPartitionSize(DecoratedKey key, long rowSize)
+    {
+        if (Guardrails.partitionSize.triggersOn(rowSize, null))
+        {
+            String what = String.format("%s.%s:%s on sstable %s",
```

Comment:
our experience is we prefer metadata.keyspace +    metadata.name +                              metadata().partitionKeyType.getString(key.getKey()) +   getFilename() to contact these strings than String.format. 
as String.format  will cost more time ,But it seems cassandra just use String.format  all time.So this comment can be ignored.

Reply:
Indeed `String.format` is everywhere. In this case, the message is only built when the guardrail is actually triggered, which shouldn't be very often.